README.md

```diff
@@ -115,28 +135,341 @@ or globally
+			public string? CreateDateDescending => ResourceManager.GetString(nameof(CreateDateDescending), CultureInfo);
+		}
+    }
+}
+```
+
+### Inner Class Visibility (per file or globally)
+
+By default inner classes are not generated, unless this setting is one of the following:
+
+ * Public
+ * Internal
+ * Private
+ * Protected
+ * SameAsOuter
+
+Case is ignored, so you could use "private".
+
+It is also possible to use "NotGenerated" to override on a file if the global setting is to generate inner classes.
+
+```xml
+<ItemGroup>
+	<EmbeddedResource Update="**/*.resx">
+		<InnerClassVisibility>private</InnerClassVisibility>
+	</EmbeddedResource>
+</ItemGroup>
+```
+
+or globally
+
+```xml
+<PropertyGroup>
+  <ResXFileCodeGenerator_InnerClassVisibility>private</ResXFileCodeGenerator_InnerClassVisibility>
+</PropertyGroup>
+```
+
+### Inner Class name (per file or globally)
+
+By default the inner class is named "Resources", which can be overriden with this setting:
+
+```xml
+<ItemGroup>
+	<EmbeddedResource Update="**/*.resx">
+		<InnerClassName>MyResources</InnerClassName>
+	</EmbeddedResource>
+</ItemGroup>
+```
+
+or globally
+
+```xml
+<PropertyGroup>
+  <ResXFileCodeGenerator_InnerClassName>MyResources</ResXFileCodeGenerator_InnerClassName>
+</PropertyGroup>
+```
+
+
+### Inner Class instance name (per file or globally)
+
+By default no instance is available of the class, but that can be made available if this setting is given.
+
+```xml
+<ItemGroup>
+	<EmbeddedResource Update="**/*.resx">
+		<InnerClassInstanceName>EveryoneLikeMyNaming</InnerClassInstanceName>
+	</EmbeddedResource>
+</ItemGroup>
+```
+
+or globally
+
+```xml
+<PropertyGroup>
+  <ResXFileCodeGenerator_InnerClassInstanceName>EveryoneLikeMyNaming</ResXFileCodeGenerator_InnerClassInstanceName>
+</PropertyGroup>
+```
+
+For brevity, settings to make everything non-static is obmitted.
+
+### Generate Code (per file or globally)
+
+By default the ancient `System.Resources.ResourceManager` is used.
 
-Linked resources namespace follow `TargetPath` if it is set.
+Benefits of using `System.Resources.ResourceManager`:
+
+* Supports custom `CultureInfo`
+* Languages are only loaded the first time a language is referenced
+* Only use memory for the languages used
+* Can ship satellite dlls seperately
+
+Disadvantages of using `System.Resources.ResourceManager`
+
+* The satellite dlls are always lazy loaded, so cold start penalty is high
+* Satellite dlls requires that you can scan the dir for which files are available, which can cause issues in some project types
+* Loading a satellite dll takes way more memory than just loading the respective strings
+* Build time for .resources -> satellite dll can be quite slow (~150msec per file)
+* Linker optimization doesn't work, since it cannot know which resources are referenced
+
+Benefits of using `VocaDB` code generation:
+
+* All languages are placed in the main dll, no more satellite dlls
+* Lookup speed is ~600% faster (5ns vs 33ns)
```

Comment:
Would you mind if I ask how to benchmark it? (I guess `launchSettings.json` is related to this?) I'm interested in it.

Reply:
I used benchmark.net and measured how long it took to do the 2 integration tests